### PR TITLE
210 bring `get_code_dependency`

### LINKED ISSUE #210: [Feature Request]: Implement `names` argument in `get_code` to subset code for specific objects.
### Feature description

technically, it means to move `teal.data:::get_code_dependencies` to `teal.code`.
We also need to deprecate `teal.data::get_code` method and just use the `get_code` method from teal.code.

----

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines.

### Security Policy

- [X] I agree to follow this project's Security Policy.

### PR DESCRIPTION
**Pull request by m7pr**
BLOCKED by 
- https://github.com/insightsengineering/teal.data/pull/340

Closes 
- #210 

Companion to 
- https://github.com/insightsengineering/teal.data/pull/343 
- https://github.com/insightsengineering/teal/pull/1388

# Description

Brings `names` parameter to `get_code` so that you can limit returned code to specific objects (and the lines that create those objects). `get_code_dependency` was moved from `teal.data`

# Tested with

``` r
library(teal.code)

# EVAL CODE
q <- qenv()
q <- eval_code(q, code = c("a <- 1", "b <- 2"))
q@code
#> [1] "a <- 1\nb <- 2"
get_code(q, names = "a")
#> [1] "a <- 1"

# WITHIN
q <- qenv()
q <- within(q, {a <- 1; b <- 2})
q@code
#> [1] "a <- 1\nb <- 2"
get_code(q, names = "a")
#> [1] "a <- 1"

# OLD TEAL.DATA
t <- teal.data::teal_data(a = 5, code = c("a <- 1", "b <- 2"))
t@code
#> [1] "a <- 1\nb <- 2"
teal.data::get_code(t, datanames = 'a')
#> Warning in .local(object, deparse, ...): get_code(datanames) was deprecated in
#> teal.data 0.6.1, use get_code(names) instead.
#> [1] "a <- 1"
```

<sup>Created on 2024-10-16 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

# Local tests

```r
> devtools::test()
ℹ Testing teal.code
✔ | F W  S  OK | Context
✔ |         12 | qenv_concat                                                 
✔ |          8 | qenv_constructor                                            
✔ |         26 | qenv_eval_code                                              
✔ |      2  60 | qenv_get_code [1.1s]                                        
✔ |         10 | qenv_get_var                                                
✔ |          7 | qenv_get_warnings                                           
✔ |         40 | qenv_join                                                   
✔ |         14 | qenv_within                                                 
✔ |         12 | utils                                                       

══ Results ══════════════════════════════════════════════════════════════════
Duration: 2.2 s

── Skipped tests (2) ────────────────────────────────────────────────────────
• This is not urgent and can be ommitted with @linksto tag. (1):
  test-qenv_get_code.R:526:3
• We will not resolve this, as this requires code evaluation. (1):
  test-qenv_get_code.R:318:3

[ FAIL 0 | WARN 0 | SKIP 2 | PASS 189 ]
````

# Local R CMD CHECK

```r
── R CMD check results ──────────────────────────────────── teal.code 0.5.0.9010 ────
Duration: 33.5s

❯ checking for future file timestamps ... NOTE
  unable to verify current time

0 errors ✔ | 0 warnings ✔ | 1 note ✖

R CMD check succeeded
```

### CODE REVIEW

File: R/qenv-get_code.R [hunk 1]
```diff
@@ -5,23 +5,97 @@
...
+#' get_code(q, names = "a")
 #'
 #' @name get_code
 #' @rdname qenv
 #' @aliases get_code,qenv-method
 #' @aliases get_code,qenv.error-method
 #'
 #' @export
-setGeneric("get_code", function(object, deparse = TRUE, ...) {
+setGeneric("get_code", function(object, deparse = TRUE, names = NULL, ...) {
```
Comment (m7pr) [inline R/qenv-get_code.R:98]: Unsure if this is needed in the Generic

Comment (gogonzo) [reply R/qenv-get_code.R:98]: Yes, otherwise you won't be able to document the `@param`.

Comment (m7pr) [reply R/qenv-get_code.R:98]: Alrighty, thanks! I started a new habit of pointing up places where I'm not 100% sure about the implementation. Based on the `review and PR` guidelines

File: tests/testthat/test-qenv_eval_code.R [hunk 2]
```diff
@@ -76,7 +76,7 @@ testthat::test_that("eval_code works with quoted code block", {
 })
 
 testthat::test_that("eval_code fails with unquoted expression", {
-  testthat::expect_error(eval_code(qenv(), a <- b), "object 'b' not found")
```
Comment (gogonzo) [inline tests/testthat/test-qenv_eval_code.R:79]: I just checked locally, error message from evaluating this code is unchanged.

Comment (m7pr) [reply tests/testthat/test-qenv_eval_code.R]: Sorry, something must have been wrong with the setup on my side. I don't see it anymore either. Reverted the removal of error messages in those tests https://github.com/insightsengineering/teal.code/pull/214/commits/8340d0c4c827f509c3bd600f8051ec0a21b6d739

**Review by gogonzo — DISMISSED** [review]
👍

**Review by gogonzo — APPROVED** [review]
checked again, everything is fine here

**Pull request merged**